### PR TITLE
Error mapping function implementation

### PR DESCRIPTION
**Pull request by angel-ayala**
The Integral values is miss-calculated, first the integral is applied to the error (accum the error values) and then is applied the gain Ki value.

Additionally an error mapping function is implemented to transform the error in another set of values, implemented in first place for a yaw angle control of a quadcopter

### CODE REVIEW

**Comment by m-lundberg** [comment]
Hello @angel-ayala !

It's been a while since I heard from you. Are you still interested in moving forward with this pull request?

**Comment by angel-ayala** [comment]
Hi @m-lundberg, Yes I'm interested, I have been a busy right now, but I'll make some space during this week to do it.
Sorry for keeping this request open.

**Comment by angel-ayala** [comment]
All merge test were ok @m-lundberg sorry for the delay.

File: simple_pid/PID.py [hunk 1]
```diff
@@ -13,6 +14,17 @@ def _clamp(value, limits):
...
+    """Transform the angle value to a [-pi, pi) range."""
+    if angle > 0:
+        if angle > math.pi:
+            return angle - 2 * math.pi
+    else:
+        if angle < -math.pi:
+            return angle + 2 * math.pi
+    return angle
+
+
```
Comment (m-lundberg) [inline simple_pid/PID.py:27]: I don't think it's the right way to go to include the mapping function in the library. The library is on purpose small and simple, and providing pre-defined mapping functions is not within the scope of this library. I think it's enough to include it as an example in the readme, but what would be even better would be to make an actual complete example under examples/ in the project root directory that a user could run and which clearly shows how the error mapping feature works. I wont make that a strict requirement for merging this PR but if you could make such an example I would really appreciate it!

File: tests/test_pid.py [hunk 2]
```diff
@@ -216,3 +216,15 @@ def update_system(C, dt):
...
+def test_error_map():
+    from simple_pid.PID import pi_clip
+
+    sp = 0.  # setpoint
+    pid = PID(1, 0, 0, setpoint=sp, sample_time=0.1)
+    PV = 5.  # process variable
+
+    # check if error value is mapped by the function
+    pid.error_map = pi_clip
+    assert pid(PV) == pi_clip(sp - PV)  # clip the error
```
Comment (m-lundberg) [inline tests/test_pid.py:230]: Nice, thanks!

**Review by m-lundberg — COMMENTED** [review]
Hello again! Sorry for taking so long to get to this! I hope you are still up for merging this PR. Please take a look at my comment if you are. Let me know.

File: simple_pid/PID.py (hunk 1 above)
Comment (angel-ayala) [reply simple_pid/PID.py]: Yes, you are right, I added the function only in the test file.

**Review by m-lundberg — APPROVED** [review]
I'm approving this now.

**Comment by m-lundberg** [comment]
And now it's merged. Thanks a lot for your contribution!

**Pull request merged**